From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ViktorHofer <viktor.hofer@microsoft.com>
Date: Wed, 19 April 2023 18:42:00 +0000
Subject: [PATCH] Eliminate prebuilts

---
 .props/Product.props | 2 +-
 1 file changed, 1 insertion(+), 1 deletions(-)

diff --git a/.props/Product.props b/.props/Product.props
index e97b6eff..6183fbb6 100644
--- a/.props/Product.props
+++ b/.props/Product.props
@@ -29,7 +29,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="$(MicrosoftSourceLinkGitHubVersion)" PrivateAssets="All" />
   </ItemGroup>
 
   <PropertyGroup>
